basic/bittwister_test.go

```diff
@@ -0,0 +1,609 @@
+			require.NoError(t, err, "Error setting bandwidth limit")
+
+			t.Log("Starting bandwidth test. It takes a while.")
+			startTime := time.Now()
+			ctx, cancel := context.WithTimeout(context.Background(), commandTimeout)
+			defer cancel()
+			output, err := iperfClient.ExecuteCommandWithContext(ctx,
+				"iperf3", "-c", iperfServerIP,
+				"-t", fmt.Sprint(int64(iperfTestDuration.Seconds())),
+				"-P", fmt.Sprint(iperfParallelClients), "--json")
+			require.NoError(t, err, "Error executing command")
+
+			elapsed := time.Since(startTime)
+			t.Logf("Test took %d seconds", int64(elapsed.Seconds()))
+
+			var iperfOutput struct {
+				End struct {
+					SumReceived struct {
+						BitsPerSecond float64 `json:"bits_per_second"`
+					} `json:"sum_received"`
+				} `json:"end"`
+			}
+			err = json.Unmarshal([]byte(output), &iperfOutput)
+			require.NoError(t, err, "Error unmarshalling JSON")
+
+			deviationPercent := math.Abs(iperfOutput.End.SumReceived.BitsPerSecond-float64(tc.targetBandwidth)) / float64(tc.targetBandwidth) * 100
+			assert.LessOrEqual(t, deviationPercent, float64(tc.tolerancePercent), "Deviation is too high")
+
+			t.Logf("Bandwidth expected: %v \tgot: %v \tdeviation: %v%% \ttolerance: %v%%",
+				formatBandwidth(float64(tc.targetBandwidth)),
+				formatBandwidth(iperfOutput.End.SumReceived.BitsPerSecond),
+				math.Round(deviationPercent),
+				tc.tolerancePercent)
+		})
+	}
+}
+
+func TestBittwister_Packetloss(t *testing.T) {
+	t.Parallel()
+	// Setup
+
+	const (
+		numOfPingPackets = 1000
+		packetTimeout    = 10 * time.Millisecond
+		commandTimeout   = 10 * time.Minute
+	)
+
+	mother, err := knuu.NewInstance("mother")
+	require.NoError(t, err, "Error creating instance")
+
+	require.NoError(t, mother.EnableBitTwister(), "Error enabling BitTwister")
+	err = mother.SetImage(gopingImage)
+	require.NoError(t, err, "Error setting image")
+
+	gopingPort := 8001
+	err = mother.SetEnvironmentVariable("SERVE_ADDR", fmt.Sprintf("0.0.0.0:%d", gopingPort))
+	require.NoError(t, err, "Error setting environment variable")
+
+	require.NoError(t, mother.AddPortTCP(gopingPort), "Error adding port")
+	require.NoError(t, mother.Commit(), "Error committing instance")
+
+	target, err := mother.CloneWithName("target")
+	require.NoError(t, err, "Error cloning instance")
+
+	executor, err := mother.CloneWithName("executor")
+	require.NoError(t, err, "Error cloning instance")
+
+	t.Cleanup(func() {
+		if os.Getenv("KNUU_SKIP_CLEANUP") == "true" {
+			t.Log("Skipping cleanup")
+			return
+		}
+
+		require.NoError(t, executor.Destroy(), "Error destroying executor instance")
+		require.NoError(t, target.Destroy(), "Error destroying target instance")
+	})
+
+	// Prepare ping executor & target
+
+	require.NoError(t, target.EnableBitTwister(), "Error enabling BitTwister")
+	require.NoError(t, target.Start(), "Error starting target instance")
+	require.NoError(t, target.WaitInstanceIsRunning(), "Error waiting for target instance to be running")
```

Comment:
`Start()` is already blocking. So there is no need to call `WaitInstanceIsRunning()` here.
In case you want to start without waiting, you can use `StartWithoutWait()`.

Reply:
oh nice, thanks for noticing it.

Reply:
Thanks, applied